dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -173,6 +171,31 @@ public Version parseWorkflowContent(String filepath, String content, Set<SourceF
         return version;
     }
 
+
+    /**
+     * A common helper method for checking for local recursive imports
+     * @param primaryDescriptorContent content of primary descriptor
+     * @param sourcefiles Set of sourcefiles to validate
+     * @param primaryDescriptorFilePath Path of primary descriptor
+     * @param validationMessageObject Object in which to store messages
+     * @return an optional validationMessageObject
+     */
+    public Optional<Map<String, String>>  reportValidationForLocalRecursiveImports(String primaryDescriptorContent, Set<SourceFile> sourcefiles,
+            String primaryDescriptorFilePath, Map<String, String> validationMessageObject) {
+        try {
+            String parent = primaryDescriptorFilePath.startsWith("/") ? new File(primaryDescriptorFilePath).getParent() : "/";
+            checkForRecursiveLocalImports(primaryDescriptorContent, sourcefiles, new HashSet<>(), parent);
+        } catch (ParseException e) {
+            LOG.error("Recursive local imports found: ", e);
+            if (validationMessageObject == null) {
```

Comment:
This feels wrong to either append the result to an existing map or create a new map -- it's not very functional.

I would suggest changing this method to just return an Optional<String>, then have the caller append to the result to the map, if present. The caller will have the key.

Reply:
fixed; now returns an Optional String